File: 2025_03/프로그래머스 - 알고리즘 고득점Kit/[완전탐색 Lv.1] 최소직사각형/최소직사각형.py
# 가로, 세로 길이를 나눠서 보지말고 긴쪽 짧은쪽으로만 정리
def solution(sizes):
  w, h = 0, 0
  for v1, v2 in sizes:
    short, long = min(v1, v2), max(v1, v2)
    w, h = max(w, short), max(h, long)

  return w*h

File: 2025_03/프로그래머스 - Lv.1/최소직사각형/최소직사각형.js
function solution(sizes) {
  const reSizes = sizes.map(([garo, sero]) => garo < sero ? [sero, garo] : [garo, sero]);

  let [w, h] = [0, 0];
  reSizes.forEach(([garo, sero]) => {
    if(garo > w) w = garo;
    if(sero > h) h = sero;
  })
  return w*h;
}